contracts/DrugItem.sol

```diff
@@ -0,0 +1,82 @@
+pragma solidity 0.5.7;
+pragma experimental ABIEncoderV2;
+
+import "openzeppelin-solidity/contracts/ownership/Secondary.sol";
+
+
+contract DrugItem is Secondary {
+
+    struct Handover {
+        Participant to;
+        uint when;
+    }
+
+    struct Participant {
+        address id;
+        ParticipantCategory category;
+    }
+
+    struct TransitConditions {
+        int8 temperature;
+        TransitCategory category;
+    }
+
+    enum ParticipantCategory { Vendor, Carrier, Pharmacy }
+    enum TransitCategory { NotApplicable, Airplane, Ship, Truck }
+
+    bytes public drugItemId;
+    address public vendor;
+    Handover[] public handoverLog;
+    mapping(bytes32 => TransitConditions) private transitConditionsLog;
+
+    constructor(bytes memory _drugItemId, address _vendor, address _to, ParticipantCategory _participantCategory)
+        public
+        notEmptyDrugItemId(_drugItemId)
+    {
+        drugItemId = _drugItemId;
+        vendor = _vendor;
+        logHandover(_to, _participantCategory);
+    }
+
+    function getHandoverCount() public view returns (uint256) {
+        return handoverLog.length;
+    }
+
+    function getLastHandover() public view returns (Handover memory) {
+        return handoverLog[handoverLog.length - 1];
+    }
+
+    function logHandover(
+        address _to,
+        ParticipantCategory _participantCategory
+    )
+        public
+        onlyPrimary
+    {
+        require(_participantCategory != ParticipantCategory.Vendor, "Drug item can't be handed over back to any vendor");
+        // solium-disable-next-line security/no-block-members
```

Comment:
This particular disable is often enabled and as long as you know what you are doing it's safe. Move it to solium config to not pollute code.